### ISSUE TITLE
Use encoding from the serializer if specified

### ISSUE THREAD

**Issue by argaen** [argaen/aiocache#7]
There is an `encoding` attr defined for serializer because in some cases it is needed for aioredis. To keep the interface clean, the encoding part should be done in the serializer layer. This will help also in the case of `memcached` implementation

**Comment by argaen** [comment]
Need to think more about it. Redis for example needs always to have an encoding when calling `get`. The encoding should be 'utf-8' in order to return a string.

The problem is that if this logic is moved to the serializers, depending on which backend is using it, it must explicitly say which encoding must use every time is being initialized...

**Comment by argaen** [comment]
This definitely has to go to the serializer side, have a look at #20. The methods become messy just because the `aiomcache` client is always working with bytes. Need to move all this logic to the serializer

**Comment by argaen** [comment]
Can't be moved 100% to serializers... the users shouldnt know which encoding the backend does support but the backend can't deal 100% with that either.

For example, the following case: Imagine we set 'utf-8' as encoding for Redis backend. It stores bytes but then we want to retrieve strings so we are encoding/decoding every time a new value is saved/retrieved. Now imagine we are using the Pickleserializer which by default it already returns bytes... a `UnicodeDecodeError` will be raised =/.

Still need to think how to get around with this. Updated a bit the logic but the problem persists.

4627e4f
